examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,414 @@
+
+# %%
+# In this case, the atlas and the fMRI image have different resolutions.
+# So we will resample the atlas to the fMRI image. It is important to do that
+# because only :class:`~nilearn.maskers.NiftiMasker` can handle the resampling
+# of the mask to the fMRI image but other methods considered here will not.
+
+resampled_atlas = resample_to_img(
+    atlas_img,
+    fmri_img,
+    interpolation="nearest",
+    copy_header=True,
+    force_resample=True,
+)
+
+mask_imgs = []
+mask_paths = []
+for idx in range(N_REGIONS):
+    mask = resampled_atlas.get_fdata() == idx
+    mask = new_img_like(
+        ref_niimg=fmri_img,
+        data=mask,
+        affine=resampled_atlas.affine,
+        copy_header=True,
+    )
+
+    mask_imgs.append(mask)
+    path = output_dir / f"mask_{idx}.nii.gz"
+    mask.to_filename(path)
+    mask_paths.append(path)
+
+# %%
+# Mask the fMRI image using NiftiMasker
+# -------------------------------------
+# Let's first see how we would typically mask the fMRI image using
+# :class:`~nilearn.maskers.NiftiMasker`. This is the most common way to extract
+# data from an fMRI image as it makes it easy to standardize, smooth, detrend,
+# etc. the data.
+#
+# We will define a function that will do that so that it's easier to use it
+# with the ``memory_profiler`` package to measure the memory usage.
+
+from nilearn.maskers import NiftiMasker
+
+
+def nifti_masker_single(fmri_path, mask_path):
+    return NiftiMasker(mask_img=mask_path).fit_transform(fmri_path)
+
+
+# %%
+# Furthermore, we can input the fmri image and the masks in two different ways:
+#
+# 1. Using the file paths
+# 2. Using the in-memory objects
+#
+# So we will measure the memory usage for both cases.
+#
+# Let's first create a dictionary to store the memory usage for each method
+
+from memory_profiler import memory_usage
+
+nifti_masker = {"single": {"path": [], "in_memory": []}}
+
+nifti_masker["single"]["path"] = memory_usage(
+    (nifti_masker_single, (fmri_path, mask_paths[0])),
+    max_usage=True,
+)
+print(
+    "Peak memory usage: with NiftiMasker, single mask, with paths:\n"
+    f"{nifti_masker['single']['path']} MiB"
+)
+
+nifti_masker["single"]["in_memory"] = memory_usage(
+    (nifti_masker_single, (fmri_img, mask_imgs[0])),
+    max_usage=True,
+)
+print(
+    "Peak memory usage: with NiftiMasker, single mask, with in-memory image:\n"
+    f"{nifti_masker['single']['in_memory']} MiB"
+)
+
+# %%
+# Masking using NiftiMasker in parallel
+# -------------------------------------
+# Now let's see how we would mask the fMRI image using multiple masks in
+# parallel using the :mod:`joblib` package.
+#
+# Let's add another key to the previous dictionary to store the memory usage
+# for this case.
+
+from joblib import Parallel, delayed
+
+
+def nifti_masker_parallel(fmri_path, mask_paths):
+    return Parallel(n_jobs=N_REGIONS)(
+        delayed(nifti_masker_single)(fmri_path, mask) for mask in mask_paths
+    )
```

Comment:
> so it should be loading all the data into memory

I think joblib is doing some memory management here because in `Parallel` by default `mmap_mode="r"`. If we change this to `mmap_mode=None` (which turns off this memory mapping), then all the "in_memory" cases fail (for NiftiMasker, numpy indexing and numpy indexing with shared memory), with the error at the bottom:

```suggestion
def nifti_masker_parallel(fmri_path, mask_paths):
    return Parallel(n_jobs=N_REGIONS, mmap_mode=None)(
        delayed(nifti_masker_single)(fmri_path, mask) for mask in mask_paths
    )
```

```bash
  File "/data/parietal/store3/work/haggarwa/miniconda3/envs/nilearnpy/lib/python3.13/site-pa
ckages/joblib/numpy_pickle.py", line 598, in load_temporary_memmap
    JOBLIB_MMAPS.add(obj.filename)
                     ^^^^^^^^^^^^
AttributeError: 'numpy.ndarray' object has no attribute 'filename'
```

Reply:
See joblib Parallel's [docstring](https://arc.net/l/quote/fkjrqazv)

Reply:
Great ! Yes, I thought NiftiMasker was calling `get_fdata()` but I'm glad to know that I was wrong :) This is a sensible handling of this !

Reply:
Maybe this error (and the explanation for it) could feature in the user guide ! But will resolve for this issue.